### PR TITLE
Fixed a crash that happens when you attempt to get the name of an XML document

### PR DESCRIPTION
**Pull request by pi1024e**
Name now returns nil instead of crashing. This was done by adding Document to the types of XML to ignore, returning early like the other XML types ineligible for a name are. 5.1 is not out yet, so I put this is 5.0.

### CODE REVIEW

File: CoreFoundation/Parsing.subproj/CFXMLInterface.c [hunk 1]
```diff
@@ -432,11 +432,9 @@ void _CFXMLNodeSetURI(_CFXMLNodePtr node, const unsigned char* URI) {
 }
 
 void _CFXMLNodeSetPrivateData(_CFXMLNodePtr node, void* data) {
-    if (!node) {
-        return;
```
Comment (xwu) [inline CoreFoundation/Parsing.subproj/CFXMLInterface.c]: As before, please remove all unrelated stylistic changes from this PR.

**Comment by pi1024e** [comment]
Did!

**Comment by xwu** [comment]
There are still unrelated changes. I assume you meant to open a PR against the master branch?

**Comment by millenomi** [comment]
@pi1024e: I appreciate that these issues are real, but there are a few problems with your approaches:

 - One issue per patch, please. The CFString changes are unrelated, can you remove them?
 - We do not take 5.0 (monthly release) patches that aren't being backported or cherry-picked from master. Can you retarget this to master?

**Comment by YOCKOW** [comment]
I feel it's a little confusing direction.
- If this PR is retargeted to master, it will be the same with #2453.
- The approach of #2453 (that still contains unrelated changes and incorrect modification though) is now the same with #2316.
- #2316 is open still.
- That is, there are/will be three PRs to fix the same issue [SR-10776](https://bugs.swift.org/browse/SR-10776).

----
@pi1024e 

One of the important things that [I said](https://github.com/apple/swift-corelibs-foundation/pull/2453#issuecomment-517996257) is
> Why don’t you review #2316 before creating a new PR?

I think @xwu also said [the same thing](https://github.com/apple/swift-corelibs-foundation/pull/2453#issuecomment-518249913):
> if you're interested in helping to resolve such an issue or disagree with how they've approached the topic, you can review their work and offer to help that person.

**Comment by pi1024e** [comment]
@millenomi Yes, this works for master too. Let me rebase it, if that is what you need.

**Comment by pi1024e** [comment]
The CFString change was merged into master already, which is why this PR is showing it in the diff case.

**Comment by pi1024e** [comment]
@YOCKOW I will be happy to help you then, if you so wish.

**Comment by millenomi** [comment]
I'm calling an audible and consolidating the effort into https://github.com/apple/swift-corelibs-foundation/pull/2316.
